fineract-core/src/main/java/org/apache/fineract/infrastructure/core/data/SqlOperator.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core.data;
+
+import static java.lang.String.format;
+import static org.apache.fineract.infrastructure.core.service.database.DatabaseSpecificSQLGenerator.alias;
+import static org.apache.fineract.infrastructure.core.service.database.DatabaseSpecificSQLGenerator.escape;
+import static org.apache.fineract.infrastructure.core.service.database.DatabaseSpecificSQLGenerator.formatValue;
+
+import jakarta.validation.constraints.NotNull;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.fineract.infrastructure.core.exception.PlatformServiceUnavailableException;
+import org.apache.fineract.infrastructure.core.service.database.DatabaseType;
+import org.apache.fineract.infrastructure.core.service.database.JdbcJavaType;
+
+@AllArgsConstructor
+@Getter
+public enum SqlOperator {
```

Comment:
I think this class is way too complicated and trying to solve a lot of things which you don't need to solve your needs.